app/uk/gov/hmrc/helptosavefrontend/config/CustomWSConfigParser.scala

```diff
@@ -84,31 +75,48 @@ class CustomWSConfigParser @Inject()(configuration: Configuration, env: Environm
         )
       )
     )
-    modded
-  }
 
-  private def mergeAllStores(config: Configuration): Configuration = {
-    mergeStores(mergeStores(config, "key"), "trust")
+    wsClientConfig
   }
 
-  private def mergeStores(config: Configuration, name: String): Configuration = {
-    val under = config.underlying
-    if (under.hasPath(s"play.ws.ssl.${name}Manager.store")) {
-      val singleStore: ConfigObject = under.getObject(s"play.ws.ssl.${name}Manager.store")
-      val stores: util.List[AnyRef] = under.getList(s"play.ws.ssl.${name}Manager.stores").unwrapped()
-      if (singleStore != null) {
-        stores.add(singleStore)
-      }
-      config.copy(underlying = config.underlying.withValue(s"play.ws.ssl.${name}Manager.stores", ConfigValueFactory.fromIterable(stores)))
-    } else config
+  private def createTrustStoreConfig(ts: TrustStoreConfig, data: String): TrustStoreConfig = {
+
+    createTempFileForData(data) match {
+      case Success(trustFile) ⇒
+        val keyStore = KeyStore.getInstance(KeyStore.getDefaultType)
+        keyStore.load(null, null)
+
+        val certs = generateCertificates(trustFile)
+        certs.foreach { cert ⇒
+          val alias = cert.asInstanceOf[X509Certificate].getSubjectX500Principal.getName
+          keyStore.setCertificateEntry(alias, cert)
+        }
+
+        keyStore.store(new FileOutputStream(trustFile.getAbsolutePath), "".toCharArray)
+        logger.info(s"Successfully wrote truststore data to file: ${trustFile.getAbsolutePath}")
+        ts.copy(filePath = Some(trustFile.getAbsolutePath), data = None)
+
+      case Failure(error) ⇒
+        logger.error(s"Error storing trust data in temp file", error)
+        sys.error(s"Error storing trust data in temp file: ${error.getMessage}")
+    }
   }
 
+  private def generateCertificates(file: File): Seq[Certificate] = {
 
-  def writeToTempFile(data: Array[Byte]): Try[File] = Try {
+    val dis = new DataInputStream(new FileInputStream(file))
+    val bytes = new Array[Byte](dis.available)
+    dis.readFully(bytes)
+    val bais = new ByteArrayInputStream(bytes)
+    val cf = CertificateFactory.getInstance("X.509")
+    cf.generateCertificates(bais).toList
```

Comment:
Doesn't
```scala
private def generateCertificates(file: File): Seq[Certificate] = {
  val fileInputStream = new FileInputStream(file)
  val factory = CertificateFactory.getInstance("X.509")
  try {
    cf.generateCertificates(bais).toList
  } finally {
   case NonFatal(_) =>
      fileInputStream.close()
     // handle error (sys.error ?)
  }
}
```
work? Do we need to create the `ByteArrayInputStream`? Also, I made sure that the resources were closed even in the case of an exception being thrown